drivers/gpio/gpio_intel_apl.c

```diff
@@ -304,6 +284,87 @@ static int gpio_intel_apl_config(struct device *dev, int access_op,
 	return 0;
 }
 
+static int gpio_intel_apl_pin_interrupt_configure(struct device *dev,
+		unsigned int pin, enum gpio_int_mode mode,
+		enum gpio_int_trig trig)
+{
+	const struct gpio_intel_apl_config *cfg = dev->config->config_info;
+	struct gpio_intel_apl_data *data = dev->driver_data;
+	u32_t raw_pin, cfg0, cfg1;
+	u32_t reg, reg_en;
+
+	/* no double-edge triggering according to data sheet */
+	if (trig == GPIO_INT_TRIG_BOTH) {
+		return -ENOTSUP;
+	}
+
+	if (pin > cfg->num_pins) {
+		return -EINVAL;
+	}
+	pin = k_array_index_sanitize(pin, cfg->num_pins + 1);
+
+	raw_pin = cfg->pin_offset + pin;
+
+	if (!check_perm(dev, raw_pin)) {
+		return -EPERM;
+	}
+
+	/* clear (by setting) interrupt status bit */
+	reg = cfg->reg_base + REG_GPI_INT_STS_BASE;
+	sys_bitfield_set_bit(reg, raw_pin);
```

Comment:
Shouldn't we clear interrupt status after new configuration is applied and shortly before enabling it? Otherwise pin will be sensitive for a while to some old interrupt configuration we have no control of.